File: tulip-standalone/demo-gradle-kotlin/app/src/main/kotlin/DemoUser.kt
import io.github.wfouche.tulip.api.TulipUser

class DemoUser(userId: Int, threadId: Int) : TulipUser(userId, threadId) {

    override fun onStart(): Boolean {
        return true
    }

    override fun action1(): Boolean {
        Thread.sleep(10)
        return true
    }

    override fun action2(): Boolean {
        Thread.sleep(20)
        return true
    }

    override fun action3(): Boolean {
        return true
    }

    override fun onStop(): Boolean {
        return true
    }
}

File: tulip-standalone/demo-gradle-scala/app/src/main/scala/org/example/DemoUser.scala
package org.example

import io.github.wfouche.tulip.api.TulipUser
import io.github.wfouche.tulip.api.TulipConsole

class DemoUser(val userId: Int, val threadId: Int) extends TulipUser(userId, threadId) {

  override def onStart(): Boolean = {
    TulipConsole.put("ScalaDemoUser " + userId)
    true
  }

  override def action1(): Boolean = {
    Thread.sleep(10)
    true
  }

  override def action2(): Boolean = {
    Thread.sleep(20)
    true
  }

  override def action3(): Boolean = true

  override def onStop(): Boolean = true

}